sdk/iot/hub/samples/src/paho_iot_hub_twin_example.c

```diff
@@ -363,98 +545,51 @@ static int send_reported_property()
     printf("Failed to publish reported property, return code %d\n", rc);
     return rc;
   }
+
   return rc;
 }
 
-int main()
+static az_result update_property(az_span desired_payload)
 {
-  int rc;
+  az_json_parser json_parser;
+  az_json_token token;
+  az_json_token_member token_member;
 
-  // Read in the necessary environment variables and initialize the az_iot_hub_client
-  if (az_failed(rc = read_configuration_and_init_client()))
-  {
-    printf("Failed to read configuration from environment variables, return code %d\n", rc);
-    return rc;
-  }
+  // Parse desired property payload
+  AZ_RETURN_IF_FAILED(az_json_parser_init(&json_parser, desired_payload));
+  AZ_RETURN_IF_FAILED(az_json_parser_parse_token(&json_parser, &token));
 
-  // Get the MQTT client id used for the MQTT connection
-  if (az_failed(
-          rc
-          = az_iot_hub_client_get_client_id(&client, mqtt_client_id, sizeof(mqtt_client_id), NULL)))
+  AZ_RETURN_IF_FAILED(az_json_parser_parse_token_member(&json_parser, &token_member));
+  while (!az_span_is_content_equal(token_member.name, version_name))
   {
-    printf("Failed to get MQTT clientId, return code %d\n", rc);
-    return rc;
-  }
+    // Update property locally if found
+    if (az_span_is_content_equal(token_member.name, reported_property_name))
+    {
+      double temp_property_value = 0.0;
+      AZ_RETURN_IF_FAILED(az_json_token_get_number(&token_member.token, &temp_property_value));
 
-  // Create the Paho MQTT client
-  if ((rc = MQTTClient_create(
-           &mqtt_client, mqtt_endpoint, mqtt_client_id, MQTTCLIENT_PERSISTENCE_NONE, NULL))
-      != MQTTCLIENT_SUCCESS)
-  {
-    printf("Failed to create MQTT client, return code %d\n", rc);
-    return rc;
+      printf("Updating \"%.*s\" locally.\n", az_span_size(reported_property_name), az_span_ptr(reported_property_name));
+      reported_property_value = (int32_t)temp_property_value;
+      return AZ_OK;
+    }
+    AZ_RETURN_IF_FAILED(az_json_parser_parse_token_member(&json_parser, &token_member));
   }
 
-  // Set the callback for incoming MQTT messages
-  if ((rc = MQTTClient_setCallbacks(mqtt_client, NULL, NULL, on_received, NULL))
-      != MQTTCLIENT_SUCCESS)
-  {
-    printf("Failed to set MQTT callbacks, return code %d\n", rc);
-    return rc;
-  }
+  printf("Did not find \"%.*s\" in desired property payload.\n", az_span_size(reported_property_name), az_span_ptr(reported_property_name));
+  return AZ_ERROR_ITEM_NOT_FOUND;
+}
+
+static void increment_property_count()
+{ 
+  printf("Incrementing \"%.*s\" locally.\n", az_span_size(reported_property_name), az_span_ptr(reported_property_name));
```

Comment:
small thing but this is nice to see on the output :+1:

Reply:
thanks.